tests/rust_usage_test/tests/integration_test.rs

```diff
@@ -263,6 +263,16 @@ mod lifetime_correctness {
         // this line should compile:
         table.get::<&'static str>(0, None);
     }
+
+    #[test]
+    fn table_object_self_lifetime_is_short_lived() {
+        // setup
+        let buf = load_file("../monsterdata_test.mon");
+        let monster = my_game::example::get_root_as_monster(&buf);
+        // This line won't compile if "self" is required to live for the lifetime of buf above as the borrow disappears at the end of the closure.
+        let enemy_of_my_enemy = monster.enemy().map(|e| e.name());
```

Comment:
This would be clearer as `monster.enemy().unwrap().name()`, WDYT?

Reply:
You are correct, but in this case I'm specifically testing a failing pattern that won't compile. It'll work if there's no closure in place as the temporary consumed from the `Option` will be allowed live to the end of the function. The closure "forces the issue".

Reply:
I could modify the test to more explicitly use `{` and `}` instead of a closure. For example, this would probably fail too:

```
         let buf = load_file("../monsterdata_test.mon");
        let monster = my_game::example::get_root_as_monster(&buf);
        // This line won't compile if "self" is required to live for the lifetime of buf above as the borrow disappears at the end of the closure.
        let enemy_of_my_enemy = {
            let enemy = monster.enemy();
            e.unwrap().name();
        }
        ...
```

Reply:
Okay, either way is fine by me. I think you should at least add more explanation that *that* particular code is designed intentionally to check a lifetime property. That will help future contributors/maintainers.

Reply:
I gave it another shot to see if I could explain better - let me know what you think of this! 😄

Reply:
@mmastrac nice!